modules/diskq/logqueue-disk-reliable.c

```diff
@@ -395,9 +396,9 @@ _free(LogQueue *s)
 {
   LogQueueDiskReliable *self = (LogQueueDiskReliable *)s;
 
-  _empty_queue(self->qreliable);
-  _empty_queue(self->qbacklog);
-  _empty_queue(self->qout);
+  _empty_queue(self, self->qreliable);
+  _empty_queue(self, self->qbacklog);
+  _empty_queue(self, self->qout);
   g_queue_free(self->qreliable);
```

Comment:
hmm can we have elements in those queues once stop() is called? I mean stop() does free these elements, doesn't it?

Maybe an assert() would be a better choice (e.g. asserting that stop was called or asserting that the queues are empty)

Reply:
You are right, in theory we should not have any messages in the memory queues at this point.